### PR TITLE
client: Firefox autoscrolling

### PR DESCRIPTION
**Pull request by bmollot**

### CODE REVIEW

**Comment by bakape** [comment]
Thanks. I'll go through and test this later today or tomorrow.

File: client/util/scroll.ts [hunk 1]
```diff
@@ -61,42 +60,35 @@ function isAtBottom(): boolean {
...
+}, { passive: true })
 
-	// Unlock from bottom, when the tab is hidden, unless set not to
-	document.addEventListener("visibilitychange", () => {
-		const opts = trigger("getOptions")
-		if (document.hidden && (opts && !opts.alwaysLock)) {
-			atBottom = false
+// Use a MutationObserver to jump to the bottom of the page when a new
+// post is made, unless the user has scrolled up from the bottom
+let threadContainer = document.getElementById("thread-container")
```
Comment (bakape) [inline client/util/scroll.ts]: No such element on board pages

File: client/util/scroll.ts [hunk 2]
```diff
@@ -61,42 +60,35 @@ function isAtBottom(): boolean {
...
 		}
 	})
-} else {
-	// Permanently hide locking UI
-	const el = document.createElement("style")
-	el.innerHTML = `#lock, #alwaysLock, label[for="alwaysLock"]{display: none;}`
-	document.head.append(el)
-}
+})
+threadObserver.observe(threadContainer, {
```
Comment (bakape) [inline client/util/scroll.ts]: This works for new post additions, but not, when an existing post is modified (for example, blockquote text mutations).

**Comment by bmollot** [comment]
Sorry about the lazy testing on my part. Identified problems are fixed.

**Comment by bakape** [comment]
Thanks.

**Pull request merged**